vars/npmExecute.groovy

```diff
@@ -0,0 +1,65 @@
+import static com.sap.piper.Prerequisites.checkScript
+import com.sap.piper.GenerateDocumentation
+import com.sap.piper.ConfigurationHelper
+import com.sap.piper.Utils
+import groovy.transform.Field
+
+@Field def STEP_NAME = getClass().getName()
+@Field Set GENERAL_CONFIG_KEYS = []
+@Field Set STEP_CONFIG_KEYS = [
+    /**
+     * Name of the docker image that should be used, in which node should be installed and configured. Default value is 'node:8-stretch'.
+     */
+    'dockerImage',
+    /**
+     * Which NPM command should be executed. Default value is 'run build'.
+     */
+    'npmCommand']
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS + [
+    /**
+     * Docker options to be set when starting the container.
+     */
+    'dockerOptions']
+/**
+ * Executes NPM commands inside a docker container.
+ * Docker image, docker options and npm commands can be specified or configured.
+ */
+@GenerateDocumentation
+void call(Map parameters = [:]) {
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters) {
+
+        final script = checkScript(this, parameters) ?: this
+
+        // load default & individual configuration
+        Map configuration = ConfigurationHelper.newInstance(this)
+            .loadStepDefaults()
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+            .use()
+
+        new Utils().pushToSWA([
+            step: STEP_NAME,
+            stepParamKey1: 'scriptMissing',
+            stepParam1: parameters?.script == null
+        ], configuration)
+
+        try {
+            if (fileExists('package.json')) {
```

Comment:
I would throw an error if the package.json does not exist instead of skipping the command

Reply:
Ah. I overlooked that the error is thrown below.
What about 
```
if(!fileExists('package.json')){
  error "[${STEP_NAME}] package.json is not found."
}
dockerExecute ...
```

Reply:
Yes your suggestion is better. Fixed with https://github.com/SAP/jenkins-library/pull/491/commits/3e2c7db7e1592d08b84bf698c1c51b747e73aa0b